src/senaite/core/content/worksheet.py

```diff
@@ -0,0 +1,1556 @@
+        or the parameter override_analyses is set to True. Analyses that are
+        incompatible with the specified instrument will remain unchanged.
+        """
+        analyses = self.getAnalyses()
+        instrument = api.get_object(instrument, default=None)
+
+        # find out the methods supported by the instrument, if any
+        supported_methods = instrument.getRawMethods() if instrument else []
+
+        total = 0
+        for an in analyses:
+
+            if not override_analyses and an.getRawInstrument():
+                # skip, no overwrite analysis if an instrument is set
+                continue
+
+            if not an.isInstrumentAllowed(instrument):
+                # skip, instrument cannot run this analysis
+                continue
+
+            # assign the instrument
+            an.setInstrument(instrument)
+            total += 1
+
+            if an.getRawMethod() in supported_methods:
+                # the analysis method is supported by this instrument
+                continue
+
+            # reset and try to assign the first supported method
+            allowed = an.getRawAllowedMethods()
+            methods = list(filter(lambda m: m in allowed, supported_methods))
+            method = methods[0] if methods else None
+            an.setMethod(method)
+
+        mutator = self.mutator("instrument")
+        mutator(self, instrument)
+        return total
+
+    # BBB: AT schema field property
+    Instrument = property(getInstrument, setInstrument)
+
+    @security.protected(permissions.View)
+    def getRemarks(self):
+        accessor = self.accessor("remarks")
+        return accessor(self)
+
+    @security.protected(permissions.ModifyPortalContent)
+    def setRemarks(self, value):
+        mutator = self.mutator("remarks")
+        mutator(self, value)
+
+    # BBB: AT schema field property
+    Remarks = property(getRemarks, setRemarks)
+
+    @security.protected(permissions.View)
+    def getResultsLayout(self):
+        accessor = self.accessor("results_layout")
+        return accessor(self)
+
+    @security.protected(permissions.ModifyPortalContent)
+    def setResultsLayout(self, value):
+        mutator = self.mutator("results_layout")
+        mutator(self, value)
+
+    # BBB: AT schema field property
+    ResultsLayout = property(getResultsLayout, setResultsLayout)
+
+    @security.protected(permissions.View)
+    def getReplacedBy(self):
+        accessor = self.accessor("replaced_by")
+        return accessor(self)
+
+    def getRawReplacedBy(self):
+        replaced_by = self.getReplacedBy()
+        if replaced_by:
+            return replaced_by.UID()
+        return None
+
+    @security.protected(permissions.ModifyPortalContent)
+    def setReplacedBy(self, value):
+        mutator = self.mutator("replaced_by")
+        mutator(self, value)
+
+    @security.protected(permissions.View)
+    def getReplacesRejectedWorksheet(self):
```

Comment:
Remove this function. It is no longer necessary after the removal of the `reject` transition and related contents. See comment for `after_reject`